Userland/Libraries/LibJS/Parser.h

```diff
@@ -300,6 +307,7 @@ class Parser {
         bool allow_super_property_lookup { false };
         bool allow_super_constructor_call { false };
         bool in_function_context { false };
+        bool in_eval_function_context { false }; // This controls if we allow new.target or not. Note that eval("return") is not allowed, so we have to have a separate state variable for eval.
```

Comment:
Main thing I'm not certain about is this new state variable. Does it need some additional plumbing?

Reply:
Since it is part of ParserState I don't see what additional plumbing would be needed.
However can this not just be function context? (I haven't looked at the spec in a while so this could definitely be wrong)

Reply:
Actually just found that `function f() { eval("return 1"); }` is not allowed so it is not the same (maybe one extra test?)

Reply:
Added the test :+1:
What I meant by "additional plumbing" is if there is any place where I need to temporally make it false, for example here:
https://github.com/SerenityOS/serenity/blob/f4f850aaf2ca300b55e14ffbf303a4b0e4f53cb7/Userland/Libraries/LibJS/Parser.cpp#L1222

Reply:
Not sure what the answer is, I think not. Let's not block this PR over a small detail like this, we'll figure it out eventually :)